rmf_websocket/src/rmf_websocket/BroadcastServer.cpp

```diff
@@ -39,10 +39,19 @@ class BroadcastServer::Implementation
   Implementation(
     const int port,
     ApiMessageCallback callback,
-    std::optional<ApiMsgType> msg_selection)
-  : _data(std::make_shared<Data>(std::move(callback),
-      std::move(msg_selection)))
+    std::optional<ApiMsgType> msg_selection,
+    const rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr
+      node_logging_interface = nullptr)
```

Comment:
```suggestion
      node_logging_interface)
```

We can avoid using default arguments in `Implementation` constructors, by choosing what the static `make` functions pass into them. So the original `make` functions implementation can pass in a `nullptr` instead